### ISSUE TITLE
[ENHANCEMENT] Allow users to console into a container by using the $JID # in addition to the hostname.

### ISSUE THREAD

**Issue by X86BSD** [BastilleBSD/bastille#701]
**Is your feature request related to a problem? Please describe.**
This was briefly discussed in https://github.com/BastilleBSD/bastille/issues/193 along with a patch (at the time) by johanhen.
This is another usability issue, not a bug. When using bastille for more than a small number of containers, using the hostname to console into the containers becomes... tedious. No one wants to type: "bastille console reallyreallylonghostnamethattakestoolongtotypeandiserrorprone". When it would be MUCH quicker and simpler and remove that pain point to simply use the JID #. "bastille console 1" is clearly easier to do than having to type in the reallyreallylonghostname.

**Describe the solution you'd like**
What I would like to see is a small modification so that users can simply type in the JID # when console'ing into a container. I'm not proposing changing the current way of typing in the hostname, I am simply suggesting to allow the user to enter either the hostname OR the JID # to console into a container.  Ex. "bastille console 1". I am actually kind of surprised this has not been brought up before. Using hostname's only imo becomes tedious pretty quickly with more than a handful of jails. This is not a high priority of course, just something I think is a pain point I would like to see removed.

**Comment by yaazkal** [comment]
Thanks for the suggestion, let me say that a name like `reallyreallylonghostnamethattakestoolongtotypeandiserrorprone` is the user's fault in the first place as Bastille never force the naming in any way :)

That said, `bastille console 1` will not work as anyone could name its jails as numbers, but maybe an option like `bastille console -j 1` will be clearer. I don't think it should necesary has to delete the use of the command with the jail name, that must be preserved.

Let's see what other people think.

**Comment by yaazkal** [comment]
No comments in a long time. I don't see any priority for this, what do you think @bmac2 @cedwards ?

**Comment by X86BSD** [comment]
Lol yes I agree that wildly long host names are the users own fault.
However that IS what the user asked you to do. Trying to console into a
long or even moderate hostbame sucks. $JID’s are much much shorter. And
make trying to console into jails much faster and easier. I’m not sure why
there hasn’t been more comment on this. This is a clear win for usability.
Which is the sole reason I brought this up. It’s just a usability
enhancement. And if we don’t do this now I’m almost certain in the future
this is going to be an issue with users imo.

Chris

**Comment by yaazkal** [comment]
Sure, thanks for commenting back.

This issue will remain open to not lost track of it, just not a priority at the moment. We will need to agree if the proposed `-j` option is the way to go.

Anyway PRs are welcome.

Regards.

**Comment by X86BSD** [comment]
No worries! Thanks for reminding me about this and keeping it active!

Chris

**Comment by tschettervictor** [comment]
I think it would be practical to have an “autocomplete” for jail names. Iocage has this. You could do “bastille console really” and I enter a jail that starts with that word, as long as that is the only one.

**Comment by tschettervictor** [comment]
Added PR #750

**Comment by bmac2** [comment]
I like the -j option.  autocomplete is cool but jail number is the suggestion and  I think we should do it.

**Comment by tschettervictor** [comment]
Cool. I think we should expand it to allow for use with other commands also.

The "set_target" function I've recently implemented would allow for this.

I'll implement it...

**Comment by tschettervictor** [comment]
My opinion is that you don't need a switch, but if you specify a number as the TARGET, then the function will associate the number with the proper jail name and set it as such.

**Comment by tschettervictor** [comment]
I have this implemented now. Just waiting on some PRs to be accepted before I open this one.

```
root@bastille:~ # bastille console 20
[jail1]:
root@jail1:~ #
```

**Comment by X86BSD** [comment]
This is exactly what I was suggesting. Just a simple way to console into a
jail using a much shorter numeric $JID vs a horrendous long host name. This
is great! I can’t thank you enough for this. Well done! I’m positive this
will have a large impact on usability for users. Hopefully it wasn’t too
much heavy lifting to accomplish. The user community thanks you!

-Chris

**Comment by tschettervictor** [comment]
```
root@bastille:~ # bastille cmd "33 35 36" echo hi
[in]:
hi
[jail2]:
hi
[lv]:
hi
```

**Comment by tschettervictor** [comment]
```
root@bastille:~ # bastille stop 33
[in]:
in: removed
root@bastille:~ # bastille start 33
Error: JID "33" not found. Is jail running?
root@bastille:~ # bastille cmd "33 35 36" echo hi
Error: JID "33" not found. Is jail running?
[jail2]:
hi
[lv]:
hi
```

**Comment by tschettervictor** [comment]
```
root@bastille:~ # bastille cmd "test jail" echo hi
test1
Multiple jails found for jail:
jail1
jail2
[test1]:
Jail is not running.
Use [-f|--force] to force start the jail.
```
autocomplete also working

**Comment by tschettervictor** [comment]
Can you test #792 ?

@X86BSD
